libs/clients/national-registry/v1/src/lib/nationalRegistryApi.ts

```diff
@@ -123,6 +123,33 @@ export class NationalRegistryApi {
     return Array.isArray(documentData) ? documentData : [documentData]
   }
 
+  public async postUserCorrection(
+    values: FamilyCorrection,
+  ): Promise<{ success: boolean; message?: string }> {
+    const response = await this.signal(
+      'CreateAndUpdateMS_Leidretting',
+      {
+        S5RequestID: '',
+        Kennitala: values.ssn,
+        Barn: values.ssnChild,
+        Nafn: values.name,
```

Comment:
Should the user name not be fetched from Þjóðskrá?
Might be safer in terms of identifying the correct requester.

Reply:
In the front end we do not allow the user to set their own name, it's using the name from the Þjóðskrá response.
It feels kinda like doing the same thing twice if I were to add a lookup for the name to the same endpoint here as we do in the front end.

Reply:
The web front-end is not the only client that can send in the response; so in terms of "securing" the correct name, it needs to be done on the back-end.

I can e.g. call the mutation with cURL and set any name I want.

Reply:
That's true, then where do we want to stop regarding the security of said information?

I can fetch the information from Þjóðskrá to verify the name, then there is another endpoint where I can verify if the logged in user is a legal guardian/parent of this child (like we do on the front end as well when displaying the children).

So I could technically add those both so we can be sure of the name and be sure that this child's SSN is part of the sender's family number.

Reply:
Yeah, according to your answer above.. then yeah, we need to verify that the childNationalId is indeed one of the child nationalIDs of the requestor.

You can see `getFamilyMemberDetails`, which is doing the same thing.

Reply:
Indeed. I will change it.